### ISSUE TITLE
Orphaned topologies or market models should not be able to make a difference in privacy setting

### ISSUE THREAD

**Issue by grdw** [quintel/etmoses#1414]
Related #1413 

Steps to reproduce:
- Create a LES
- Update the Topology
- Save it. 
- Give the Topology the orphaned user (or delete it from the interface)
- Go back to your LES. 
- It breaks.

**Comment by ChaelKruip** [comment]
After some discussion with @grdw I think the following would fix the current issues without too much work:

Whenever a **private** topology (or market model) is deleted, warn the user about which other LESses use it too: 

"The topology you are trying to delete is used in the following LESses: LES_1, LES_2, LES_403. Are you sure you want to delete the topology **and the related LESes**?"

If the user elects to press "OK" after that, the topology and the LESses are deleted.

Whenever a **public** topology (or market model) is deleted, warn the user as follows:

"The topology you are trying to delete is used by the following LESses owned by you: ES_1, LES_2, LES_403. Are you sure you want to delete the topology **and the related LESes**?
NOTE: the topology is also used by other peoples LESses and a copy will be kept so they won't break."

If the user chooses "OK", the topology (or market model) can be assigned a new owner "public@quintel.com" which keeps all public topologies which are still used by LESes but are deleted by their original owner.

@antw what do you think about this approach?

**Comment by antw** [comment]
I liked #839 as a solution to this. Every LES would have it's own *copy* of the LES and market model, and this would allow us to get rid of the topology and market model pages entirely. There wouldn't have to be such a thing as deleting a topology or market model; to a user they'd just be part of the LES and not separate entities (like asset lists or selected strategies).

**Comment by ChaelKruip** [comment]
> I liked #839 as a solution to this. Every LES would have it's own copy of the LES and market model, and this would allow us to get rid of the topology and market model pages entirely. There wouldn't have to be such a thing as deleting a topology or market model; to a user they'd just be part of the LES and not separate entities (like asset lists or selected strategies).

I still like #839 as well. My feeling is that it is too big of an overhaul to do if we want to deploy to LIVE by the end of this week, however. @antw do you think it is comparable in effort?

**Comment by antw** [comment]
> @antw do you think it is comparable in effort?

No, but I don't think this is such a big issue that it *needs* to be resolved before deploying. I'd rather we wait and do it right, than waste effort on something we'll (hopefully?) replace soon.

If it's a deal-breaker we can prevent users from deleting a topology which is in use by one or more LESes. That's an ugly solution too, but at least it's fast to implement.

**Comment by ChaelKruip** [comment]
> No, but I don't think this is such a big issue that it needs to be resolved before deploying.

I agree with that. This takes a very specific series of actions to crop up and will not inhibit the work-flow of users at all. 

@MarjoleinSchwachofer @ThomBuijs do you agree as well?

**Comment by ThomBuijs** [comment]
Yes I agree
